spec/licensee/projects/github_project_spec.rb

```diff
@@ -0,0 +1,140 @@
+RSpec.describe Licensee::Projects::GitHubProject do
+  describe '#initialize' do
+    subject(:instance) { described_class.new(github_url) }
+
+    context 'with a GitHub URI' do
+      let(:github_url) { 'https://github.com/benbalter/licensee' }
+
+      it 'should set @repo' do
+        expect(instance.repo).to eq('benbalter/licensee')
+      end
+    end
+
+    context 'with a GitHub git URI' do
+      let(:github_url) { 'https://github.com/benbalter/licensee.git' }
+
+      it 'should set @repo, stripping the trailing extension' do
+        expect(instance.repo).to eq('benbalter/licensee')
+      end
+    end
+
+    context 'with a non-GitHub URI' do
+      let(:github_url) { 'https://gitlab.com/benbalter/licensee' }
```

Comment:
Just making a note, I know GitLab uses Licensee, if they/anyone would like more flexibility here, glad to add it, as long as we can do so safely. :smile: